src/bonds.cpp

```diff
@@ -328,6 +376,77 @@ void PeriodicDihedral::setEnergyFunction(const ParticleVector& particles) {
     };
 }
 
+void HarmonicDihedral::from_json(const Faunus::json& j) {
+    half_force_constant = j.at("k").get<double>() * 1.0_kJmol / std::pow(1.0_rad, 2) / 2.0;
+    equilibrium_dihedral = j.at("deq").get<double>() * 1.0_deg;
+}
+
+void HarmonicDihedral::to_json(Faunus::json& j) const {
+    j = {{"k", 2 * half_force_constant / (1.0_kJmol / std::pow(1.0_rad, 2))}, {"deq", equilibrium_dihedral / 1.0_deg}};
+    _roundjson(j, 6);
+}
+
+int HarmonicDihedral::numindex() const { return 4; }
+
+HarmonicDihedral::HarmonicDihedral(double k, double deq, const std::vector<int>& indices)
```

Comment:
better to use more descriptive argument names, `k` --> `force_constant` and `deq` --> `equilibrium_dihedral` (names can be the same as member names)

Reply:
Fixed this particular instance and all others with f1feccc.